src/formatDate.js

```diff
@@ -50,7 +50,58 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fmFormatSeparator = fromFormat[fromFormat.length - 1];
+  const toFormatSeparator = toFormat[toFormat.length - 1];
+  const dateArray = date.split(fmFormatSeparator);
+
+  const changedDate = [];
+
+  let year, month, day;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      fromFormat[i] = dateArray[i];
```

Comment:
1. Sorry, don't understand the reason to do it. why?
2. Write pure function, don't mutate initial values comes as arguments

Reply:
I agree, it was useless, solved.